### PR TITLE
Distributed mesh generator works with mesh adaptivity

### PR DESCRIPTION
**Pull request by fdkong**
`DistributedRectilinearMeshGenerator` is a great way to generate a big mesh for parallel simulation. It is beneficial to add the mesh adaptivity for that mesh generator. The mesh adaptivity could save the simulation cost for certain applications.

### CODE REVIEW

File: modules/phase_field/test/tests/grain_growth/tests [hunk 1]
```diff
@@ -100,10 +100,20 @@
...
+    cli_args ='UserObjects/voronoi/type=FauxPolycrystalVoronoi'
     issues = '#14697'
   [../]
 
+  [./mesh_adaptivity]
+    type = 'Exodiff'
+    input = 'voronoi_adaptivity.i'
+    exodiff = 'voronoi_adaptivity_out.e'
+    min_parallel = 2
+    requirement = 'Distributed mesh generator should work with the mesh adaptivity when solving phase field equations'
```
Comment (lindsayad) [inline modules/phase_field/test/tests/grain_growth/tests:112]: I'd recommend rewording this to say:

"The system shall be able to apply mesh adaptivity and solve phase field equations on a mesh generated in parallel."

File: test/tests/meshgenerators/distributed_rectilinear_mesh_generator/tests [hunk 2]
```diff
@@ -66,4 +66,11 @@
     cli_args = 'Mesh/gmg/num_cores_for_partition=10 Mesh/gmg/dim=3 Mesh/gmg/nx=20 Mesh/gmg/ny=20 Mesh/gmg/nz=20 Outputs/hide="pid npid" '
     requirement = 'MOOSE shall error out if the number of cores for partition is larger than the total number of cores.'
   [../]
+
+  [./distributed_rectilinear_mesh_generator_mesh_adaptivity]
+    type = 'Exodiff'
+    input = 'distributed_rectilinear_mesh_generator_adaptivity.i'
+    exodiff = 'distributed_rectilinear_mesh_generator_adaptivity_out.e'
+    requirement = 'Distributed mesh generator should work with the mesh adaptivity'
```
Comment (lindsayad) [inline test/tests/meshgenerators/distributed_rectilinear_mesh_generator/tests:74]: "The system shall be able to adapt a mesh generated in parallel."

**Review by lindsayad — CHANGES_REQUESTED** [review]
Just the requirements suggestions, otherwise this looks good to me.

**Comment by fdkong** [comment]
It is ready @lindsayad

**Review by lindsayad — APPROVED** [review]
beautiful code

**Pull request merged**